File: vane-core/src/main/java/org/oddlama/vane/core/command/enums/WeatherValue.java
package org.oddlama.vane.core.command.enums;

public enum WeatherValue {
    clear(false, false),
    sun(false, false),
    rain(true, false),
    thunder(true, true);

    private boolean isStorm;
    private boolean isThunder;

    private WeatherValue(boolean isStorm, boolean isThunder) {
        this.isStorm = isStorm;
        this.isThunder = isThunder;
    }

    public boolean storm() {
        return isStorm;
    }

    public boolean thunder() {
        return isThunder;
    }
}

File: vane-core/src/main/kotlin/org/oddlama/vane/core/command/enums/WeatherValue.kt
package org.oddlama.vane.core.command.enums

/**
 * Named weather presets mapped to storm and thunder flags.
 *
 * @param storm whether the world should be storming.
 * @param thunder whether the world should have thunder.
 */
enum class WeatherValue(val storm: Boolean, val thunder: Boolean) {
    Clear(false, false),
    Sun(false, false),
    Rain(true, false),
    Thunder(true, true);

    /**
     * Lowercase identifier used for command parsing and completions.
     */
    val displayName: String get() = name.lowercase()
}
